08_flask_html_basics.md

```diff
@@ -0,0 +1,592 @@
+
+You will need to add a `<title>My Hello App</title>` element nested under `<head>`. You should end up with something similar to this:
+
+```html
+<head>
+    <title>My Hello App</title>
+</head>
+```
+
+</details>
+
+### Adding Text and Headings
+
+> All the following content should be added under the `<body>` element, which aggregates all the content on the HTML page.
+
+Text is one of the most relevant types of content that can be added to a web page. You can add plain text to several HTML elements, which will transform it accordingly.
+
+[Headings](https://www.w3schools.com/html/html_headings.asp) are used to differentiate between different types of texts, offer a logical structuring of the web page, and can be used to provide a common style for your text elements (e.g. fonts, colours, sizes). Essentially, they act as titles and subtitles. They are denoted by `<h1>`, `<h2>`, to `<h6>` tags. By default, the text size is the largest with `h1`, and the smallest with `h6`, but this can be changed.
+
+There's a whole theory behind efficiently using headings for [Search Engine Optimization (SEO)](https://developers.google.com/search/docs/fundamentals/seo-starter-guide), but some rules of thumb are:
+- Use `<h1>` for the page's title. You shouldn't have more that 1 or 2 `<h1>` tags, and they should contain keywords relevant for your page
+- Use `<h2>` and `<h3>` for subtitles
+- Use `<h4>`, `<h5>` and `<h6>` to structure your text
+
+While the above criteria is intentionally a bit vague and there are no hard rules for the three points, they serve as a guideline for making your page more readable.
+
+### Exercise 8.4: Adding some text
+
+Add proper title and subtitle for your page. You should:
+- _Add a title_ such as "Hello from my Python app!", using an appropriate heading. Note that this is not the page's title as displayed when you hover over the tab in your browser (i.e. the `<title>` element) - it's just a text that will act as the page's title for your content, a role similar to an article's title.
+- _Add a subtitle_ such as "Here is my Python Zoo", using an appropriate heading
+
+<details markdown="1"><summary>Click here to expand an example</summary>
+
+```html
+<body>
+    <h1>Hello from my Python app!</h1>
+    <h2>Here is my Python Zoo</h2>
+</body>
+```
+
+</details>
+
+
+### Images and Links
+
+Images and links are vital for a site's appearance and functionality. You can add an image using the [`<img>`](https://www.w3schools.com/tags/tag_img.asp) tag, and a link using the [`<a>`](https://www.w3schools.com/tags/tag_a.asp) tag.
+
+So far the elements and tags were fairly simple, but these 2 are a bit more complex: they do not work properly unless you also add _attributes_ to them.
+
+> HTML attributes act similar to variables, and are used by elements in order to enable certain functionality. They are key/value pairs added to the start tag of an element, with the value being enclosed in quotes. 
```

Comment:
I've just explained that the attribute is true/false if the attribute is mentioned/omitted. I don't think it's a good idea to mention that it can also have values, as this might be unnecessarily confusing.